securedrop_client/gui/widgets.py

```diff
@@ -64,31 +133,54 @@ def on_refresh_clicked(self):
         Called when the refresh button is clicked.
         """
         self.controller.sync_api()
+        self.refresh.setIcon(load_icon('refresh_active.svg')) # temp solution to show icon clicked
+
+    def enable_refresh(self):
+        """
+        Enable the refresh button.
+        """
+        self.refresh.setEnabled(True)
+
+    def disable_refresh(self):
+        """
+        Disable the refresh button.
+        """
+        self.refresh.setEnabled(False)
 
     def _on_sync_event(self, data):
         """
         Called when the refresh call completes
         """
-        self.refresh.setEnabled(data != 'syncing')
+        self.refresh.setIcon(load_icon('refresh.svg')) # temp solution to show icon clicked
 
-    def show_message(self, message, duration=0):
+    def _on_error_status_timeout_event(self):
+        self.error_bar.hide()
+        self.error_icon.hide()
+        self.error_status_bar.hide()
+
+    def update_activity_status(self, message: str, duration: int):
         """
-        Display a status message to the user. Optionally, supply a duration
+        Display an activity status message to the user. Optionally, supply a duration
         (in milliseconds), the default will continuously show the message.
         """
-        self.showMessage(message, duration)
+        self.activity_status_bar.showMessage(message, duration)
 
-    def hide_refresh_icon(self):
+    def update_error_status(self, message: str, duration: int):
         """
-        Hide refresh icon.
+        Display an activity status message to the user. Optionally, supply a duration
+        (in milliseconds), the default will continuously show the message.
         """
-        self.refresh.hide()
+        self.error_status_bar.showMessage(message, duration)
+        self.error_status_timer.start(duration)
```

Comment:
The error status widget is timed to popup for the same amount of time as we display the message.